File: PUGD-frontend/graphql/mutations/acquisition/provider.js
import gql from 'graphql-tag';

const InsertProvider = gql`
  mutation(
    $establishement: String!
    $name: String!
    $account: String!
    $adress: String!
    $phone: String!
    $email: String!
    $website: String!
  ) {
    insertProvider(
      establishement: $establishement
      name: $name
      account: $account
      adress: $adress
      phone: $phone
      email: $email
      website: $website
    )
  }
`;

const UpdateProvider = gql`
  mutation(
    $_id: String!
    $establishement: String!
    $name: String!
    $account: String!
    $adress: String!
    $phone: String!
    $email: String!
    $website: String!
  ) {
    updateProvider(
      _id: $_id
      establishement: $establishement
      name: $name
      account: $account
      adress: $adress
      phone: $phone
      email: $email
      website: $website
    )
  }
`;

const DeleteProvider = gql`
  mutation($_id: String!) {
    deleteProvider(_id: $_id)
  }
`;

//important
module.exports = {
  InsertProvider,
  DeleteProvider,
  UpdateProvider
};

File: PUGD-api/models/AcquisitionModels/Provider.go
package AcquisitionModels

import "go.mongodb.org/mongo-driver/bson/primitive"

type Provider struct {
	Id             primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	Establishement string             `json:"establishement"`
	Name           string             `json:"name"`
	Account        string             `json:"account"`
	Adress         string             `json:"adress"`
	Phone          string             `json:"phone"`
	Email          string             `json:"email"`
	Website        string             `json:"website"`
}
